amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -0,0 +1,219 @@
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class AmplifyTools implements Plugin<Project> {
+    void apply(Project project) {
+        def doesNodeExist = true
+        def doesGradleConfigExist
+
+        def profile = 'default'
+        def accessKeyId = null
+        def secretAccessKey = null
```

Comment:
We need to support session-based temporary credentials as well, not just static and profile

Reply:
@undefobj do you have thoughts on using session-based temporary credentials with this plugin

Reply:
The way to do this is with an IAM role which uses a profile. We basically need to validate that what the CLI supports is supported here as well: https://docs.amplify.aws/cli/usage/iam-roles-mfa

Reply:
Another problem here is that we're relying on a profile called `default` which may or may not exist.

Reply:
Thats only the placeholder profile name. If the customer is not using a profile named default (or secret key and access key) they should specify a different profile name in amplify-gradle-config.json and the plugin will pick up that profile

Reply:
Okay, but we'll need to document that. Thanks.